Verifier Docker image does not publish to local maven

SCC Version: `2.2.0.RELEASE`

I am experimenting with the offline mode of contracts and fetching them from local .m2 folder, which is mounted as a volume to a SCC Verifier docker container.

I would expect that the resulting provider stubs would be published to the same local repository. However, when the `PUBLISH_ARTIFACTS` environment variable is set to `true`, the stubs are published only to a remote repository.

**Reproducibility steps**
Try running contract tests with configuration similar to:
```
  contract-tests:
    image: springcloud/spring-cloud-contract:2.2.0.RELEASE
    environment:
      APPLICATION_BASE_URL: http://example-app:8080
      PROJECT_GROUP: example.stubs
      PROJECT_NAME: testProject
      PROJECT_VERSION: "1.0"
      PUBLISH_ARTIFACTS: "true"
      EXTERNAL_CONTRACTS_WORK_OFFLINE: "true"
      EXTERNAL_CONTRACTS_GROUP_ID: example
      EXTERNAL_CONTRACTS_ARTIFACT_ID: contracts
      EXTERNAL_CONTRACTS_VERSION: "1.0-SNAPSHOT"
      EXTERNAL_CONTRACTS_PATH: example-app
```

It will fail with:
```
[DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Daemon worker: released lock on root.1
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':publishStubsPublicationToMavenRepository'.
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Failed to publish publication 'stubs' to repository 'maven'
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]    > Could not GET 'http://localhost:8081/artifactory/libs-release-local/example/stubs/testProject/1.0-SNAPSHOT/maven-metadata.xml'.
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]       > Connect to localhost:8081 [localhost/127.0.0.1] failed: Connection refused (Connection refused)
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]  Run with --scan to get full insights.
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Exception is:
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':publishStubsPublicationToMavenRepository'.
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.accept(ExecuteActionsTaskExecuter.java:166)
...
```

**Fix proposals**
Have `mavenLocal()` repository always present in the publication repositories.

Dealing with remote publishing:
1. When `EXTERNAL_CONTRACTS_WORK_OFFLINE` is set to `true` simply allow only publishing to local maven repository. This could however break some backwards compatibility and removes the option of publishing the stubs to remote repo after using local contracts.
or
2. Add additional property such as `PUBLISH_ARTIFACTS_OFFLINE`. If `PUBLISH_ARTIFACTS` is set to `true`, then the user has the option to allow only offline publish by setting `PUBLISH_ARTIFACTS_OFFLINE` to `true`.

--- a/docker/spring-cloud-contract-docker/project/build.gradle
+++ b/docker/spring-cloud-contract-docker/project/build.gradle
@@ -109,11 +109,14 @@ test.finalizedBy("copyOutput")
 
 publishing {
 	repositories {
-		maven {
-			url getProp('REPO_WITH_BINARIES_URL') ?: 'http://localhost:8081/artifactory/libs-release-local'
-			credentials {
-				username getProp('REPO_WITH_BINARIES_USERNAME') ?: 'admin'
-				password getProp('REPO_WITH_BINARIES_PASSWORD') ?: 'password'
+		mavenLocal()
+		if (!Boolean.parseBoolean(getProp("PUBLISH_ARTIFACTS_OFFLINE") ?: "false")) {
+			maven {
+				url getProp('REPO_WITH_BINARIES_URL') ?: 'http://localhost:8081/artifactory/libs-release-local'
+				credentials {
+					username getProp('REPO_WITH_BINARIES_USERNAME') ?: 'admin'
+					password getProp('REPO_WITH_BINARIES_PASSWORD') ?: 'password'
+				}
 			}
 		}
 	}

--- a/docs/src/main/asciidoc/docker-project.adoc
+++ b/docs/src/main/asciidoc/docker-project.adoc
@@ -85,6 +85,8 @@ which is the default URL of https://jfrog.com/artifactory/[Artifactory] running 
 - `REPO_WITH_BINARIES_USERNAME`: (optional) Username when the Artifact Manager is secured. Defaults to `admin`.
 - `REPO_WITH_BINARIES_PASSWORD`: (optional) Password when the Artifact Manager is secured. Defaults to `password`.
 - `PUBLISH_ARTIFACTS`: If set to `true`, publishes the artifact to binary storage. Defaults to `true`.
+- `PUBLISH_ARTIFACTS_OFFLINE`: If set to `true` and the publishing is enabled, it will publish the artifacts to local
+`.m2` only. Defaults to `false`.
 
 These environment variables are used when contracts lay in an external repository. To enable
 this feature, you must set the `EXTERNAL_CONTRACTS_ARTIFACT_ID` environment variable.

